app/src/main/java/com/cronus/zdone/settings/SettingsView.kt

```diff
@@ -44,18 +43,30 @@ class SettingsView(context: Context) : BaseScreenView<SettingsScreen>(context) {
     }
 
     fun setApiKey(token: String) {
+        apiKeyEditText.removeTextChangedListener(apiKeyWatcher)
         apiKeyEditText.setText(token)
+        apiKeyEditText.post { apiKeyEditText.addTextChangedListener(apiKeyWatcher) }
     }
 
     fun setWorkTime(defaultWorkTime: Int) {
-        workTimeMinsEditText.setText(defaultWorkTime.toString())
+        setWorkTimeTextNoListener(defaultWorkTime.toString())
         if (workTimeMinsEditText.hasFocus()) {
-            workTimeMinsEditText.setSelection(workTimeMinsEditText.text.length)
+            workTimeMinsEditText.post { workTimeMinsEditText.setSelection(workTimeMinsEditText.text.length) }
```

Comment:
use `post` to ensure that the action happens after any previous methods applied to this view. `post` adds the runnable to the end of the message queue for the main thread